frontend-ssr/components/lessons/LessonComments/newComment/NewComment.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { RootState } from '~/redux/root.reducer';
+
+import { addComment } from '~/redux/lesson-comments/lesson-comments.actions';
+import { Comment } from '~/redux/lesson-comments/types';
+
+import { uuid } from '~/services/Utils';
+import CommentBlock from '../comment/CommentBlock';
+
+import styles from './NewComment.module.scss';
+
+interface State {
+  newComments: Comment[];
+}
+
+class NewComment extends React.Component<ConnectedProps<typeof connector>, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newComments: [],
+    };
+  }
+
+   handleSubmit = (e) => {
+     e.preventDefault();
+
+     const { dispatch } = this.props;
+     const { newComments } = this.state;
+
+     const form = e.target;
+     const username = e.target.username.value;
+     const name = e.target.name.value;
+     const avatar = e.target.avatar.value;
+     const comment = e.target.comment.value;
+
+     const newComment = {
+       _id: uuid(),
+       avatar,
+       username,
+       name,
+       comment,
+       timestamp: Date.now(),
+     };
+
+     dispatch(addComment(newComment));
+
+     this.setState({
+       newComments: [...newComments, newComment],
+     });
+
+     form.reset();
+   };
+
+   render() {
+     const { newComments } = this.state;
+
+     return (
+       <div className={styles['new-comment']}>
+         {newComments.map((newComment) => <CommentBlock key={newComment._id} {...newComment} />)}
+         <form onSubmit={this.handleSubmit}>
+           <div>
+             <input
```

Comment:
Regarding the inputs let's just have a textarea with the actual comment for now as we don't need the rest of the inputs.

For this story let's mock the UI, avatar and so on. For the final integration things will look like this:

1. User wants to comment
2. If it's already logged in - comment gets added
3. Otherwise, show login popup. User logs in/registers -> then the comment is added.